src/game_battle.cpp

```diff
@@ -232,25 +233,36 @@ void Game_Battle::NextTurn(Game_Battler* battler) {
 	Main_Data::game_party->IncTurns();
 }
 
-void Game_Battle::UpdateGauges() {
+void Game_Battle::UpdateAtbGauges() {
 	std::vector<Game_Battler*> battlers;
-	Main_Data::game_enemyparty->GetActiveBattlers(battlers);
-	Main_Data::game_party->GetActiveBattlers(battlers);
+	Main_Data::game_enemyparty->GetBattlers(battlers);
+	Main_Data::game_party->GetBattlers(battlers);
 
-	int max_agi = 1;
+	const auto use_2k3e_algo = Player::IsRPG2k3E();
 
-	for (std::vector<Game_Battler*>::const_iterator it = battlers.begin();
-		it != battlers.end(); ++it) {
-		max_agi = std::max(max_agi, (*it)->GetAgi());
+	int sum_agi = 0;
+	for (auto* bat: battlers) {
+		// RPG_RT uses dead and state restricted battlers to contribute to the sum.
+		if (!bat->IsHidden()) {
+			sum_agi += bat->GetAgi();
+		}
 	}
-
-	// Only affects how fast the gauges move, can be safely clamped
-	max_agi = Utils::Clamp(max_agi, 1, 1000);
-
-	for (std::vector<Game_Battler*>::const_iterator it = battlers.begin();
-		it != battlers.end(); ++it) {
-		if (!(*it)->GetBattleAlgorithm() && (*it)->CanAct()) {
-			(*it)->UpdateGauge(1000 / max_agi);
+	sum_agi *= 100;
+
+	const int max_atb = Game_Battler::GetMaxAtbGauge();
+
+	for (auto* bat: battlers) {
+		// RPG_RT always updates atb for non-hidden enemies, even if they can't act.
+		if (bat->Exists() && (bat->CanAct() || bat->GetType() == Game_Battler::Type_Enemy))
+		{
+			const auto agi = bat->GetAgi();
+			auto increment = max_atb / (sum_agi / (agi + 1));
```

Comment:
Thats one of the cases where I would discourage the use of "auto". The return type of GetAgi is hidden and the type of increment is unknown (int or float)

Reply:
You can look at it 2 ways. On one hand the type is hidden. On the other hand, this expression works cleanly no matter what the type of agi is. If agi was an int, it works, if it was a double it also works.

If you made it `const int agi = bat->GetAgi()` and then the battler agi changed to double, you would get a silent cast. That's arguably a bug and a hard to find one at that.

It's trivial either way for this case.